[BUG] Several files are using utf-8 encoding, should use ACSII

**Describe the bug**
https://github.com/opensearch-project/opensearch-build/issues/635#issuecomment-929706299

These files cause build failure due to encoding not in ASCII:
```
src/main/java/org/opensearch/ad/caching/CacheBuffer.java: UTF-8 Unicode text
src/main/java/org/opensearch/ad/caching/PriorityCache.java: UTF-8 Unicode text
src/main/java/org/opensearch/ad/caching/PriorityTracker.java: UTF-8 Unicode text
src/main/java/org/opensearch/ad/model/Entity.java: UTF-8 Unicode text
src/main/java/org/opensearch/ad/ratelimit/ColdEntityWorker.java: UTF-8 Unicode text
src/main/java/org/opensearch/ad/ratelimit/RateLimitedRequestWorker.java: UTF-8 Unicode text
src/main/java/org/opensearch/ad/model/ADTaskState.java: Java source, UTF-8 Unicode text
src/main/java/org/opensearch/ad/model/InitProgressProfile.java: Java source, UTF-8 Unicode text
```

```
/tmp/tmpbcap9s69/anomaly-detection/src/main/java/org/opensearch/ad/caching/CacheBuffer.java:54: error: unmappable character (0x80) for encoding US-ASCII
 * We use a layered cache to manage active entities??? states.  We have a two-level
                                                    ^
/tmp/tmpbcap9s69/anomaly-detection/src/main/java/org/opensearch/ad/caching/CacheBuffer.java:54: error: unmappable character (0x99) for encoding US-ASCII
 * We use a layered cache to manage active entities??? states.  We have a two-level
                                                     ^
/tmp/tmpbcap9s69/anomaly-detection/src/main/java/org/opensearch/ad/caching/CacheBuffer.java:56: error: unmappable character (0xE2) for encoding US-ASCII
 * dedicated cache that stores ten (dynamically adjustable) entities??? states per
 * ```

**Other plugins installed**
Please mention if you are using this plugin along side any other plugin. Security for example.

**To Reproduce**
Run the build on Jenkins with `./gradlew assemble --no-daemon --refresh-dependencies -DskipTests=true -Dopensearch.version=1.1.0 -Dbuild.snapshot=false` on CentOS7.
Not producible on local only on Jenkins.

**Expected behavior**
Javadoc build correctly without this unmapping issues.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Desktop (please complete the following information):**
 - OS: [e.g. iOS]
 - Browser [e.g. chrome, safari]
 - Version [e.g. 22]

**Additional context**
Add any other context about the problem here.

--- a/build.gradle
+++ b/build.gradle
@@ -58,10 +58,13 @@ plugins {
 }
 
 tasks.withType(JavaCompile) {
-	options.encoding = "UTF-8"
+    options.encoding = "UTF-8"
 }
 tasks.withType(Test) {
-	systemProperty "file.encoding", "UTF-8"
+    systemProperty "file.encoding", "UTF-8"
+}
+tasks.withType(Javadoc) {
+    options.encoding = 'UTF-8'
 }
 
 repositories {

